skbio/metadata/tests/test_interval.py

```diff
@@ -0,0 +1,568 @@
+                     metadata={'name': 'sagA', 'function': 'transport'})
+
+        self.assertTrue(f._interval_metadata is not None)
+        self.assertListEqual(f.intervals, [(1, 2), (4, 7)])
+        self.assertListEqual(f.boundaries, [(True, False), (False, False)])
+        self.assertDictEqual(f.metadata, {'name': 'sagA',
+                                          'function': 'transport'})
+
+    def test_scrambled_constructor(self):
+        f = Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[(4, 7), (1, 2)],
+                     boundaries=[(True, False), (False, True)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+
+        self.assertTrue(f._interval_metadata is not None)
+        self.assertListEqual(f.intervals, [(1, 2), (4, 7)])
+        self.assertListEqual(f.boundaries, [(False, True), (True, False)])
+        self.assertDictEqual(f.metadata, {'name': 'sagA',
+                                          'function': 'transport'})
+
+    def test_bad_constructor(self):
+        with self.assertRaises(TypeError):
+            Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[1, (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+
+    def test_bad_constructor_bad_boundaries(self):
+        with self.assertRaises(ValueError):
+            Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[(1, 2), (4, 7)],
+                     boundaries=[(False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+
+    def test_repr_one_interval(self):
+        im = IntervalMetadata()
+        f = Interval(interval_metadata=im,
+                     intervals=[(1, 2)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        exp = (r"<Interval: start=1, end=2, 1 contiguous interval,"
+               " 2 metadata keys, dropped=False>")
+        res = repr(f)
+        self.assertEqual(res, exp)
+
+    def test_repr_two_intervals(self):
+        im = IntervalMetadata()
+        f = Interval(interval_metadata=im,
+                     intervals=[(1, 2), (4, 7)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        exp = (r"<Interval: start=1, end=7, 2 non-contiguous intervals,"
+               " 2 metadata keys, dropped=False>")
+        res = repr(f)
+        self.assertEqual(res, exp)
+
+    def test_str_two_intervals(self):
+        im = IntervalMetadata()
+        f = Interval(interval_metadata=im,
+                     intervals=[(1, 2), (4, 7)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+        exp = (r"<Interval: start=1, end=7, 2 non-contiguous intervals,"
+               " 2 metadata keys, dropped=False>")
+        res = str(f)
+        self.assertEqual(res, exp)
+
+    def test_cmp(self):
```

Comment:
thanks! I realize this object is not implementing the `__cmp__` spec (well, python 2 spec...), but is it correct for `cmp(foo, foo)` to return `False`? If the implementation here is as desired, and "correct" for its use, would you consider revising the doc string associated with `_cmp` to explicitly state that this method does not check for equality? As I understand it, cmp methods generally return a negative, zero, or a positive integer (e.g., C++ [qsort](http://en.cppreference.com/w/cpp/algorithm/qsort))